mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java

```diff
@@ -426,8 +427,10 @@ private SourceStorageBatchClient getSourceStorageBatchClient(OkapiConnectionPara
   private void postProcessMarcRecord(Record record, InitialRecord rawRecord) {
     String matchedId = getValue(record, TAG_999, 's');
     if (StringUtils.isNotBlank(matchedId)) {
-      record.setMatchedId(matchedId);
-      record.setGeneration(null); // in case the same record is re-imported, generation should be calculated on SRS side
+      if (hasIndicator(record, 's')) {
```

Comment:
It would be good if these magic string like 's' or 'i' to be saved in a final variable and reused in other locations. It will also convey more meaning based on its more descriptive name.

Reply:
Totally agree, I've made changes for that.